caf-audit-service/src/main/java/com/hpe/caf/services/audit/server/api/impl/AuditeventsApiServiceImpl.java

```diff
@@ -0,0 +1,429 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.TimeZone;
+import java.util.UUID;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JaxRSServerCodegen", date = "2017-04-28T07:15:58.947+01:00")
+public class AuditeventsApiServiceImpl extends AuditeventsApiService {
+
+    private static final Logger LOG = LogManager.getLogger(AuditeventsApiServiceImpl.class.getName());
+
+    private static final String ERR_MSG_APPLICATION_ID_NOT_SPECIFIED = "The application identifier has not been specified";
+    private static final String ERR_MSG_PROCESS_ID_NOT_SPECIFIED = "The process identifier has not been specified";
+    private static final String ERR_MSG_THREAD_ID_NOT_SPECIFIED = "The thread identifier has not been specified";
+    private static final String ERR_MSG_EVENT_ORDER_NOT_SPECIFIED = "The event order has not been specified";
+    private static final String ERR_MSG_EVENT_TIME_NOT_SPECIFIED = "The event time has not been specified";
+    private static final String ERR_MSG_EVENT_TIME_SOURCE_NOT_SPECIFIED = "The event time source has not been specified";
+    private static final String ERR_MSG_USER_ID_NOT_SPECIFIED = "The user identifier has not been specified";
+    private static final String ERR_MSG_TENANT_ID_NOT_SPECIFIED = "The tenant identifier has not been specified";
+    private static final String ERR_MSG_CORRELATION_ID_NOT_SPECIFIED = "The correlation identifier has not been specified";
+    private static final String ERR_MSG_EVENT_TYPE_ID_NOT_SPECIFIED = "The event type identifier has not been specified";
+    private static final String ERR_MSG_EVENT_CATEGORY_ID_NOT_SPECIFIED = "The event category identifier has not been specified";
+    private static final String ERR_MSG_CUSTOM_FIELDS_NOT_SPECIFIED = "Custom audit event fields have not been specified";
+    private static final String ERR_MSG_ES_HOST_AND_PORT_MISSING = "The Elasticsearch host and port have not been provided";
+
+    @Override
+    public Response auditeventsPost(NewAuditEvent newAuditEvent,SecurityContext securityContext) throws NotFoundException {
+        //  Index new audit event into Elasticsearch.
+        try {
+            LOG.debug("Start indexing audit event message into Elasticsearch");
+            AddNewAuditEvent(newAuditEvent);
+            LOG.debug("Indexing audit event message into Elasticsearch complete");
+            return Response.noContent().build();
+        } catch (BadRequestException | ConfigurationException e){
+            return Response.status(Response.Status.BAD_REQUEST).entity(e.getMessage()).type("text/plain").build();
+        } catch(Exception e){
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getMessage()).type("text/plain").build();
```

Comment:
During a face to face review with Connor we discussed when to return a 400 Bad Request and a 500 Internal Server Error.  Connor is going to investigate this a little further with regard to what we should respond with when there is a ConfigurationException and when there is a Date ParsingException or a  NumberFormatException